### PR TITLE
Add DiffIgnoreWitespaceEol and deprecate DiffIgnoreWitespaceEol

### LINKED ISSUE #773: Typo in "DiffIgnoreWitespaceEol"
The constant `DiffIgnoreWitespaceEol` contains a typo, and should be named `DiffIgnoreWhitespaceEol` (like it's libgit2-counterpart).

https://github.com/libgit2/git2go/blame/b1eec9a4662a2e64953aa83365f84f5c773b6399/diff.go#L434

I'll send a backwards compatible patch, with the missing "h" added.

### PR DESCRIPTION
**Pull request by zegl**
DiffIgnoreWitespaceEol contains a typo and does not have the same name as it's libgit2 counterpart.

Fixes #773

### CODE REVIEW

File: diff.go [hunk 1]
```diff
@@ -487,7 +487,9 @@ const (
 
 	DiffIgnoreWhitespace       DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE
 	DiffIgnoreWhitespaceChange DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_CHANGE
-	DiffIgnoreWitespaceEol     DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	DiffIgnoreWhitespaceEol    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	// Deprecated: use DiffIgnoreWhitespaceEol instead
+	DiffIgnoreWitespaceEol DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
```
Comment (suhaibmujahid) [inline diff.go:492]: I feel that the following looks more explicit that there is no difference between the two of them. What do you think?
```suggestion
	DiffIgnoreWitespaceEol = DiffIgnoreWhitespaceEol
```

**Review by suhaibmujahid — COMMENTED** [review]
Thank you @zegl, good catch.

File: diff.go (hunk 1 above)
Comment (zegl) [reply diff.go]: Agreed, I've updated the patch. :)

**Review by suhaibmujahid — APPROVED** [review]

File: diff.go (hunk 1 above)
Comment (suhaibmujahid) [reply diff.go]: Thank you.

File: diff.go [hunk 2]
```diff
@@ -487,7 +487,9 @@ const (
 
 	DiffIgnoreWhitespace       DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE
 	DiffIgnoreWhitespaceChange DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_CHANGE
-	DiffIgnoreWitespaceEol     DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	DiffIgnoreWhitespaceEol    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	// Deprecated: use DiffIgnoreWhitespaceEol instead
+	DiffIgnoreWitespaceEol = DiffIgnoreWhitespaceEol
```
Comment (lhchavez) [inline diff.go:492]:
```suggestion
	DiffIgnoreWitespaceEOL = DiffIgnoreWhitespaceEol
```

per https://github.com/golang/go/wiki/CodeReviewComments#initialisms, the convention in Go is to use uppercase for initialisms.

File: diff.go [hunk 3]
```diff
@@ -487,7 +487,9 @@ const (
 
 	DiffIgnoreWhitespace       DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE
 	DiffIgnoreWhitespaceChange DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_CHANGE
-	DiffIgnoreWitespaceEol     DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	DiffIgnoreWhitespaceEol    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	// Deprecated: use DiffIgnoreWhitespaceEol instead
+	DiffIgnoreWitespaceEOL = DiffIgnoreWhitespaceEol
```
Comment (lhchavez) [inline diff.go:492]:
```suggestion
	DiffIgnoreWhitespaceEOL    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
	// Deprecated: use DiffIgnoreWhitespaceEol instead
	DiffIgnoreWitespaceEol = DiffIgnoreWhitespaceEOL
```

File: diff.go [hunk 4]
```diff
@@ -487,7 +487,9 @@ const (
 
 	DiffIgnoreWhitespace       DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE
 	DiffIgnoreWhitespaceChange DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_CHANGE
-	DiffIgnoreWitespaceEol     DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	DiffIgnoreWhitespaceEOL    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	// Deprecated: use DiffIgnoreWhitespaceEol instead
```
Comment (lhchavez) [inline diff.go:491]:
```suggestion
	// Deprecated: use DiffIgnoreWhitespaceEOL instead
```

File: diff.go [hunk 5]
```diff
@@ -487,7 +487,9 @@ const (
 
 	DiffIgnoreWhitespace       DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE
 	DiffIgnoreWhitespaceChange DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_CHANGE
-	DiffIgnoreWitespaceEol     DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	DiffIgnoreWhitespaceEOL    DiffOptionsFlag = C.GIT_DIFF_IGNORE_WHITESPACE_EOL
+	// Deprecated: use DiffIgnoreWhitespaceEOL instead
+	DiffIgnoreWitespaceEol = DiffIgnoreWhitespaceEOL
```
Comment (lhchavez) [inline diff.go:492]: Can this be moved to `deprecated.go`? that way everything that's deprecated is in the same file.

**Review by lhchavez — COMMENTED** [review]
I'll take a look at the CI failures :/

**Comment by lhchavez** [comment]
CI should be fixed by https://github.com/libgit2/git2go/pull/776 !

**Pull request merged**